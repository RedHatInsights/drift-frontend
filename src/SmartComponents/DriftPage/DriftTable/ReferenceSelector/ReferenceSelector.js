import React, { Component } from 'react';
import { Tooltip } from '@patternfly/react-core';
import { OutlinedStarIcon, StarIcon } from '@patternfly/react-icons';
import PropTypes from 'prop-types';

class ReferenceSelector extends Component {
    constructor(props) {
        super(props);

        this.state = {
            isReference: this.props.isReference
        };
    }

    renderIcon() {
        const { updateReferenceId, item } = this.props;
        const { isReference } = this.state;

        return (
            isReference
                ? <StarIcon
                    className='reference-selector pointer'
                    data-ouia-component-type='PF4/Button'
                    data-ouia-component-id={ 'reference-selector-' + item.id }
                    onClick={ () => updateReferenceId() }
                />
                : <OutlinedStarIcon
                    className='reference-selector pointer'
                    data-ouia-component-type='PF4/Button'
                    data-ouia-component-id={ 'reference-selector-' + item.id }
                    onClick={ () => updateReferenceId(item.id) }
                />
        );
    }

    renderMessage() {
        const { isReference } = this.state;
        const { item } = this.props;
        let type = item.type;

        if (item.type === 'historical-system-profile') {
            type = 'historical system';
        }

        if (isReference) {
            return <div>This is the reference the other items are being compared against.</div>;
        } else {
            return <div>Use this { type } as a reference to compare.</div>;
        }
    }

    render() {
        return (
            <Tooltip
                position='top'
                content={ this.renderMessage() }
            >
                { this.renderIcon() }
            </Tooltip>
        );
    }
}

ReferenceSelector.propTypes = {
    isReference: PropTypes.bool,
    updateReferenceId: PropTypes.func,
    item: PropTypes.object
};

export default ReferenceSelector;
