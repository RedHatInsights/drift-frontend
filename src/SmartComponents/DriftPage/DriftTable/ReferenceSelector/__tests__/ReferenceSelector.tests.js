import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import ReferenceSelector from '../ReferenceSelector';

describe('ReferenceSelector', () => {
    let props;

    beforeEach(() => {
        props = {
            isReference: false,
            id: 'abcd1234',
            updateReferenceId: jest.fn()
        };
    });

    it('should render correctly', () => {
        const wrapper = shallow(
            <ReferenceSelector { ...props } />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render mount correctly with isReference false', () => {
        const wrapper = mount(
            <ReferenceSelector { ...props } />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render mount correctly with isReference true', () => {
        props.isReference = true;

        const wrapper = mount(
            <ReferenceSelector { ...props } />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    describe('API', () => {
        it('should set isReference false by props', () => {
            const wrapper = mount(
                <ReferenceSelector { ...props } />
            );

            expect(wrapper.state('isReference')).toEqual(false);
        });
    });

    it('should set isReference true by props', () => {
        props.isReference = true;

        const wrapper = mount(
            <ReferenceSelector { ...props } />
        );

        expect(wrapper.state('isReference')).toEqual(true);
    });

    it('should call updatedReferenceId with id', () => {
        const wrapper = mount(
            <ReferenceSelector { ...props } />
        );

        wrapper.find('OutlinedStarIcon').simulate('click');
        expect(props.updateReferenceId).toHaveBeenCalledWith('abcd1234');
    });

    it('should call updatedReferenceId with undefined', () => {
        props.isReference = true;

        const wrapper = mount(
            <ReferenceSelector { ...props } />
        );

        wrapper.find('StarIcon').simulate('click');
        expect(props.updateReferenceId).toHaveBeenCalledWith();
    });
});
