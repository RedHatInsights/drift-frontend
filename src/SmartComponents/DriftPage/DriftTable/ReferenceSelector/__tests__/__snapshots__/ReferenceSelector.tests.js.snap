// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReferenceSelector should render correctly 1`] = `
<Tooltip
  content={
    <div>
      Use this 
       as a reference to compare.
    </div>
  }
  position="top"
>
  <OutlinedStarIcon
    className="reference-selector pointer"
    color="currentColor"
    noVerticalAlign={false}
    onClick={[Function]}
    size="sm"
  />
</Tooltip>
`;

exports[`ReferenceSelector should render mount correctly with isReference false 1`] = `
<ReferenceSelector
  isReference={false}
  item={
    Object {
      "id": "abcd1234",
    }
  }
  updateReferenceId={[MockFunction]}
>
  <Tooltip
    content={
      <div>
        Use this 
         as a reference to compare.
      </div>
    }
    position="top"
  >
    <Popper
      appendTo={[Function]}
      distance={15}
      enableFlip={true}
      flipBehavior={
        Array [
          "top",
          "right",
          "bottom",
          "left",
          "top",
          "right",
          "bottom",
        ]
      }
      isVisible={false}
      onBlur={[Function]}
      onDocumentClick={false}
      onDocumentKeyDown={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTriggerEnter={[Function]}
      placement="top"
      popper={
        <div
          className="pf-c-tooltip"
          id="pf-tooltip-1"
          role="tooltip"
          style={
            Object {
              "maxWidth": null,
              "opacity": 0,
              "transition": "opacity 300ms cubic-bezier(.54, 1.5, .38, 1.11)",
            }
          }
        >
          <TooltipArrow />
          <TooltipContent
            isLeftAligned={false}
          >
            <div>
              Use this 
               as a reference to compare.
            </div>
          </TooltipContent>
        </div>
      }
      popperMatchesTriggerWidth={false}
      positionModifiers={
        Object {
          "bottom": "pf-m-bottom",
          "left": "pf-m-left",
          "right": "pf-m-right",
          "top": "pf-m-top",
        }
      }
      trigger={
        <OutlinedStarIcon
          aria-describedby="pf-tooltip-1"
          className="reference-selector pointer"
          color="currentColor"
          noVerticalAlign={false}
          onClick={[Function]}
          size="sm"
        />
      }
      zIndex={9999}
    >
      <FindRefWrapper
        onFoundRef={[Function]}
      >
        <OutlinedStarIcon
          aria-describedby="pf-tooltip-1"
          className="reference-selector pointer"
          color="currentColor"
          noVerticalAlign={false}
          onClick={[Function]}
          size="sm"
        >
          <svg
            aria-describedby="pf-tooltip-1"
            aria-hidden={true}
            aria-labelledby={null}
            className="reference-selector pointer"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            role="img"
            style={
              Object {
                "verticalAlign": "-0.125em",
              }
            }
            viewBox="0 0 576 512"
            width="1em"
          >
            <path
              d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
            />
          </svg>
        </OutlinedStarIcon>
      </FindRefWrapper>
    </Popper>
  </Tooltip>
</ReferenceSelector>
`;

exports[`ReferenceSelector should render mount correctly with isReference true 1`] = `
<ReferenceSelector
  isReference={true}
  item={
    Object {
      "id": "abcd1234",
    }
  }
  updateReferenceId={[MockFunction]}
>
  <Tooltip
    content={
      <div>
        This is the reference the other items are being compared against.
      </div>
    }
    position="top"
  >
    <Popper
      appendTo={[Function]}
      distance={15}
      enableFlip={true}
      flipBehavior={
        Array [
          "top",
          "right",
          "bottom",
          "left",
          "top",
          "right",
          "bottom",
        ]
      }
      isVisible={false}
      onBlur={[Function]}
      onDocumentClick={false}
      onDocumentKeyDown={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTriggerEnter={[Function]}
      placement="top"
      popper={
        <div
          className="pf-c-tooltip"
          id="pf-tooltip-2"
          role="tooltip"
          style={
            Object {
              "maxWidth": null,
              "opacity": 0,
              "transition": "opacity 300ms cubic-bezier(.54, 1.5, .38, 1.11)",
            }
          }
        >
          <TooltipArrow />
          <TooltipContent
            isLeftAligned={false}
          >
            <div>
              This is the reference the other items are being compared against.
            </div>
          </TooltipContent>
        </div>
      }
      popperMatchesTriggerWidth={false}
      positionModifiers={
        Object {
          "bottom": "pf-m-bottom",
          "left": "pf-m-left",
          "right": "pf-m-right",
          "top": "pf-m-top",
        }
      }
      trigger={
        <StarIcon
          aria-describedby="pf-tooltip-2"
          className="reference-selector pointer"
          color="currentColor"
          noVerticalAlign={false}
          onClick={[Function]}
          size="sm"
        />
      }
      zIndex={9999}
    >
      <FindRefWrapper
        onFoundRef={[Function]}
      >
        <StarIcon
          aria-describedby="pf-tooltip-2"
          className="reference-selector pointer"
          color="currentColor"
          noVerticalAlign={false}
          onClick={[Function]}
          size="sm"
        >
          <svg
            aria-describedby="pf-tooltip-2"
            aria-hidden={true}
            aria-labelledby={null}
            className="reference-selector pointer"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            role="img"
            style={
              Object {
                "verticalAlign": "-0.125em",
              }
            }
            viewBox="0 0 576 512"
            width="1em"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            />
          </svg>
        </StarIcon>
      </FindRefWrapper>
    </Popper>
  </Tooltip>
</ReferenceSelector>
`;
